nautobot/extras/jobs.py

```diff
@@ -85,7 +85,15 @@ class RunJobTaskFailed(Exception):
     """Celery task failed for some reason."""
 
 
-class BaseJob(Task):
+class JobMeta(type):
+    """Job metaclass to automatically register subclasses in celery."""
+
+    def __init__(cls, name, bases, dct):
+        if not cls.__dict__.get("abstract", False):
```

Comment:
We should document this as a feature -- apps like SSOT will want/need to be able to define their own abstract job subclasses as well, and now that job registration has changed from opt-in to opt-out this will need to be handled by them.

Reply:
I wonder if we should update this to look for `nautobot.extras.Job` in `bases` so that the abstract attr isn't required?

Reply:
That'd be closer to the existing behavior but I like the idea of flagging things explicitly rather than relying on base class selection to inform magic behavior (I think `abstract` would actually make it so we can get rid of `BaseJob` class entirely in a later release?)

Reply:
Sounds like a good plan to me. Should we consider this a breaking change and push it back to 3.0 or target to next for 2.1?